### PR TITLE
feat(iota-transaction-builder): add more doc comments

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Add more doc comments to functions that had none

## Links to any relevant issues

https://github.com/iotaledger/iota/issues/3722

## Type of change

- Documentation Fix

### CODE REVIEW

File: crates/iota-transaction-builder/src/lib.rs [hunk 1]
```diff
@@ -631,6 +640,8 @@ impl TransactionBuilder {
         Ok(args)
     }
 
+    /// Build a TransactionKind::ProgrammableTransaction that contains
```
Comment (Alex6323) [inline crates/iota-transaction-builder/src/lib.rs]: put the enum variant in brackets as well?

Comment (Thoralf-M) [reply crates/iota-transaction-builder/src/lib.rs]: Done

**Review by Alex6323 — APPROVED** [review]
lgtm

**Review by valeriyr — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Review by Alex6323 — APPROVED** [review]

File: crates/iota-transaction-builder/src/lib.rs [hunk 2]
```diff
@@ -212,6 +215,8 @@ impl TransactionBuilder {
         ))
     }
 
+    /// Add a TransferObjects command to the provided
```
Comment (DaughterOfMars) [inline crates/iota-transaction-builder/src/lib.rs]: Want to add the link here?

Comment (Thoralf-M) [reply crates/iota-transaction-builder/src/lib.rs]: Done

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**